cli/parse_args.ts

```diff
@@ -450,316 +473,250 @@ export function parseArgs<
 
       let broken = false;
-      for (const [j, letter] of letters.entries()) {
+      for (let j = 0; j < letters.length; j++) {
         const next = arg.slice(j + 2);
 
         if (next === "-") {
-          setArg(letter, next, arg);
+          setArgument(letters[j], next, arg, true);
           continue;
         }
 
-        if (/[A-Za-z]/.test(letter) && next.includes("=")) {
-          setArg(letter, next.split(/=(.+)/)[1], arg);
+        if (/[A-Za-z]/.test(letters[j]) && /=/.test(next)) {
+          setArgument(letters[j], next.split(/=(.+)/)[1], arg, true);
           broken = true;
           break;
         }
 
         if (
-          /[A-Za-z]/.test(letter) &&
+          /[A-Za-z]/.test(letters[j]) &&
           /-?\d+(\.\d*)?(e-?\d+)?$/.test(next)
         ) {
-          setArg(letter, next, arg);
+          setArgument(letters[j], next, arg, true);
           broken = true;
           break;
         }
 
-        if (letters[j + 1]?.match(/\W/)) {
-          setArg(letter, arg.slice(j + 2), arg);
+        if (letters[j + 1] && letters[j + 1].match(/\W/)) {
+          setArgument(letters[j], arg.slice(j + 2), arg, true);
           broken = true;
           break;
-        } else {
-          setArg(letter, get(flags.strings, letter) ? "" : true, arg);
         }
+        setArgument(
+          letters[j],
+          stringSet.has(letters[j]) ? "" : true,
+          arg,
+          true,
+        );
       }
 
-      const key = arg.at(-1)!;
+      key = arg.slice(-1);
       if (!broken && key !== "-") {
         const nextArg = args[i + 1];
         if (
           nextArg &&
           !/^(-|--)[^-]/.test(nextArg) &&
-          !get(flags.bools, key) &&
-          (get(aliases, key) ? !aliasIsBoolean(key) : true)
+          !booleanSet.has(key) &&
+          (aliasMap.get(key)
+            ? !aliasIsBoolean(aliasMap, booleanSet, key)
+            : true)
         ) {
-          setArg(key, nextArg, arg);
+          setArgument(key, nextArg, arg, true);
           i++;
-        } else if (nextArg && (nextArg === "true" || nextArg === "false")) {
-          setArg(key, nextArg === "true", arg);
+        } else if (nextArg && isBooleanString(nextArg)) {
+          const value = parseBooleanString(nextArg);
+          setArgument(key, value, arg, true);
           i++;
         } else {
-          setArg(key, get(flags.strings, key) ? "" : true, arg);
+          setArgument(key, stringSet.has(key) ? "" : true, arg, true);
         }
       }
-    } else {
-      if (!flags.unknownFn || flags.unknownFn(arg) !== false) {
-        argv._.push(flags.strings["_"] ?? !isNumber(arg) ? arg : Number(arg));
-      }
-      if (stopEarly) {
-        argv._.push(...args.slice(i + 1));
-        break;
-      }
+      continue;
     }
-  }
 
-  for (const [key, value] of Object.entries(defaults)) {
-    if (!hasKey(argv, key.split("."))) {
-      setKey(argv, key, value, false);
+    if (unknownFn?.(arg) !== false) {
```

Comment:
This does the same as the removed code, but more readable with the use of Set.

Reply:
I don't follow. Is this comment correct?

Reply:
Updated comment